### DISCUSSION TITLE
Pulse counter

### DISCUSSION THREAD

**Discussion by geraldFS** [arendst/Tasmota#12600] [Q&A] [answered]
Hello!

I would like to use an input on an ESP8266 as a pulse generator.

I am currently using the counter, but it counts up every click and does not reset itself.
With it would be enough if every impulse (on / off) is transmitted via MQTT.

The encoder is a reed switch that I am currently debouncing with debounceLow 1000.

alternatively it would work if I can get a rule that always sets the counter to zero when the chip reconnects.

Although I would much prefer a source of inspiration.

greetings

Gerald

**Comment by barbudor** [comment]
Not sure to understand your use case.
What exactly are you measuring ? 
What is the frequency of events ?
What information are you expecting Tasmota to provide and at which rate ?

**Comment by sfromis** [comment, marked as answer]
Not clear what you mean by "pulse generator" and counter in the same context.

You can get a separate message for every pulse (or transition) by configuring the input as a Switch, and selecting a Switchmode to match (or maybe just use SetOption114). Not sure how fast impulses can be for this to work reliably.

You can get a counter to reset at TelePeriod using SetOption79.

**Comment by geraldFS** [comment]
Sorry, I was too cryptic again.

Here is a brief description of the application.

I operate a screw conveyor on the heating system, for more this one rotates at about 0.8 revolutions / minute, this should be monitored.
Ie. A pulse is generated about every minute - this is to be transmitted to a registration unit via MQTT.
There the consumption is calculated and processed accordingly.

I am currently using the counter, which works wonderfully so far, except for the restriction that it always transfers the total value when restarted, so the result is no longer correct.
But that's exactly what I can fix with SetOption79. I actually didn't find that.
Think that this will solve my problem already.

**Reply by sfromis** [reply]
Well, SetOption79 will reset the counter at every TelePeriod, which by default is 5 minutes. If you actually want the counter incrementing longer term, you could reset the counter value when you deem this to be relevant, possibly including at boot time (or maybe disable TelePeriod reporting). If you just want pulses, without having Tasmota count, the switch config I suggested may be simpler.

Or you could also use a counter, and then have a rule doing a publish whenever the counter changes value.
